From 0b31cd37f55169fcade984279f61f5d3d98ac068 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 21 Jul 2017 12:07:44 +0800
Subject: [PATCH 2251/5242] MLK-16046: mx8 imaging SS: Fix build warning

commit  c659592209af5cba7fbbf7bac6216f1c7145b084 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix build warning:
warning: missing braces around initializer [-Wmissing-braces]

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-hw.c |   14 +++++++-------
 drivers/media/platform/imx8/mxc-isi-hw.h |    3 ---
 2 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-isi-hw.c b/drivers/media/platform/imx8/mxc-isi-hw.c
index d209b87..14477f8 100644
--- a/drivers/media/platform/imx8/mxc-isi-hw.c
+++ b/drivers/media/platform/imx8/mxc-isi-hw.c
@@ -63,7 +63,7 @@ static void dump_isi_regs(struct mxc_isi_dev *mxc_isi)
 }
 #endif
 
-struct colorspace_coeffs coeffs[] = {
+static const u32 coeffs[2][6] = {
 	/* A2,A1,      B1, A3,     B3, B2,      C2, C1,     D1, C3,   D3, D2   */
 	/* YUV2RGB */
 	{ 0x0000012A, 0x012A0198, 0x0730079C, 0x0204012A, 0x01F00000, 0x01800180 },
@@ -239,12 +239,12 @@ void mxc_isi_channel_set_csc(struct mxc_isi_dev *mxc_isi)
 	printk_pixelformat("output fmt", dst_fmt->fourcc);
 
 	if (mxc_isi->cscen) {
-		writel(coeffs[csc].coeff[0], mxc_isi->regs + CHNL_CSC_COEFF0);
-		writel(coeffs[csc].coeff[1], mxc_isi->regs + CHNL_CSC_COEFF1);
-		writel(coeffs[csc].coeff[2], mxc_isi->regs + CHNL_CSC_COEFF2);
-		writel(coeffs[csc].coeff[3], mxc_isi->regs + CHNL_CSC_COEFF3);
-		writel(coeffs[csc].coeff[4], mxc_isi->regs + CHNL_CSC_COEFF4);
-		writel(coeffs[csc].coeff[5], mxc_isi->regs + CHNL_CSC_COEFF5);
+		writel(coeffs[csc][0], mxc_isi->regs + CHNL_CSC_COEFF0);
+		writel(coeffs[csc][1], mxc_isi->regs + CHNL_CSC_COEFF1);
+		writel(coeffs[csc][2], mxc_isi->regs + CHNL_CSC_COEFF2);
+		writel(coeffs[csc][3], mxc_isi->regs + CHNL_CSC_COEFF3);
+		writel(coeffs[csc][4], mxc_isi->regs + CHNL_CSC_COEFF4);
+		writel(coeffs[csc][5], mxc_isi->regs + CHNL_CSC_COEFF5);
 	}
 
 	writel(val, mxc_isi->regs + CHNL_IMG_CTRL);
diff --git a/drivers/media/platform/imx8/mxc-isi-hw.h b/drivers/media/platform/imx8/mxc-isi-hw.h
index f32c1e6..062f777 100644
--- a/drivers/media/platform/imx8/mxc-isi-hw.h
+++ b/drivers/media/platform/imx8/mxc-isi-hw.h
@@ -450,9 +450,6 @@ enum isi_csi_coeff {
 	YUV2RGB = 0,
 	RGB2YUV,
 };
-struct colorspace_coeffs {
-	u32	coeff[6];
-};
 
 void mxc_isi_channel_init(struct mxc_isi_dev *mxc_isi);
 void mxc_isi_channel_deinit(struct mxc_isi_dev *mxc_isi);
-- 
1.7.9.5

